Lithops compatibility issues with AWS Lambda

aitorarjona opened the pull request:
In this PR I've changed some of the Lithops base code in order to make it compatible with AWS Lambda compute backend.

AWS Lambda has two main limitations that make it not compatible with Lithops when using nested Lithops executions (lithops in lithops):

1. It cannot use multiprocessing because `/dev/shm`, used to share memory between processes, is [not available in lambda](https://aws.amazon.com/blogs/compute/parallel-processing-in-python-with-aws-lambda/).
2. `$HOME` is mounted on a read-only filesystem.

This break the following code:

- Using ThreadPool in [`wait_storage.py`](https://github.com/lithops-cloud/lithops/blob/f8e572f074c6d376ffb012661c147a7200ea389e/lithops/wait/wait_storage.py#L25): This has been solved replacing multiprocessing with concurrent.futures.
- Using `multiprocessing.Queue` in [`Invoker class`](https://github.com/lithops-cloud/lithops/blob/f8e572f074c6d376ffb012661c147a7200ea389e/lithops/invoker.py#L76): This has been solved by replacing multiprocessing.Queue with queue.Queue when Lithops is running inside a serverless function.
- Cannot save runtime meta to local cache in [`storage/storage.py`](https://github.com/lithops-cloud/lithops/blob/f8e572f074c6d376ffb012661c147a7200ea389e/lithops/storage/storage.py#L316), which raises and exception and then it is assumed that the runtime is not deployed, when it is: This has been solved wrapping the write to file code in a try/catch block that doesn't write to file if it can't, but still returns the runtime meta retrieved form Object Storage. In fact, if Lithops is running in a serverless function, there is no need to write to file because it will be erased anyways.


Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

gilv commented on setup.py:
```diff
@@ -29,11 +29,35 @@
+        'google-auth',
+        'azure-storage-blob==2.1.0',
+        'azure-storage-queue==2.1.0'
     ],
```
@aitorarjona @JosepSampe we already discussed it in the past. We should avoid that setup install all those dependencies, azure, amazon, google, etc. It should be per backend may be or configuration options in setup, so it's not installed by default at least, only if requested, like setup.py azure, etc.

aitorarjona replied:
How about moving these dependencies to `extras_require` in `setup.py`?
Then `pip install lithops` would only install required dependencies and, for example, to install Google Cloud backend it would be `pip install lithops[gcp]`.

aitorarjona commented:
I'm closing this PR until #378 is resolved, since merging both would produce conflicts.

aitorarjona commented:
Updates:
- Added alternate backends dependencies as `extra_require` in `setup.py`. Updated docs with an extra step to install dependencies.
- Updated knative Python 3.8 container image: Previous container image didn't had gunicorn installed.

All tests passing.

JosepSampe commented:
@gilv I's all ok for me. If you don't have any objections I will merge it

gilv commented:
@JosepSampe let me  review it a bit

gilv commented on runtime/knative/Dockerfile.python38:
```diff
@@ -1,32 +1,22 @@
-FROM python:3.8-slim-buster
```
@aitorarjona @JosepSampe why only Python 3.8 need to be changed? Is this 3.8 specific?

aitorarjona replied:
Python 3.8 Dockerfile was missing gunicorn in `RUN pip install ....` layer, so `CMD exec gunicorn --bind :$PORT --workers 1 --timeout 600 lithopsproxy:proxy` would fail when running a container with this image. I just copy-pasted the Python3.7 Dockerfile and changed `FROM python:3.7-slim-buster` to `FROM python:3.8.2-slim-buster`.
